lib/modules/platform/gerrit/index.md

```diff
@@ -0,0 +1,65 @@
+# Gerrit
+
+## Authentication
+
+First, ![HTTP access token](/assets/images/gerrit-http-password.png) for the bot account.
+
+Let Renovate use your HTTP access token by doing _one_ of the following:
+
+- Set your HTTP access token as a `password` in your `config.js` file
+- Set your HTTP access token as an environment variable `RENOVATE_PASSWORD`
+- Set your HTTP access token when you run Renovate in the CLI with `--password=`
+
+Make sure this user is allowed to assign the Code-Review label with "+2" to his own changes or "automerge" can't work.
+
+Remember to set `platform=gerrit` somewhere in your Renovate config file.
+
+## Renovate PR/Branch-Model with Gerrit and needed Permissions
+
+If you use the "Code-Review" label and want `automerge` working, then you have to enable `gerritAutoApprove=true` in your renovate config.
+In this case the bot will automatically add the _Code-Review_ label with the value "+2" to each created "pull-request" (Gerrit-Change).
+
+_Important: The login should be allowed to give +2 for the Code-Review label._
+
+**Additionally, the following two settings are mandatory. It will not work without these settings.**
+
+```
+{
+  platformCommit: true, //allow to reuse the Change-Id
+  gitNoVerify: ["push"], //allow-commit-hook to generate a Change-Id
+}
+```
+
+**Current Workarounds/Changes to git-based-commands:**
+
+They should be moved behind the Platform interface (optionally) to avoid such workarounds.
+Perhaps it should be possible in the future to integrate a platform which is not based on git.
+
+- registerBranch()
+  A new method which allows to register virtual branches (not really exists) including commitSHA and whether they have been "modified" from another user. (see isBranchModified below)
+
+- isBranchModified()
+  Checks if the last uploader was different to the renovate user and store this into "branchIsModified[branchname]". For Gerrit we rely on the above pre-registration in initRepo().
+
+- isBranchConflicted()
+  This on tries to merge "origin/${branchName}" into baseBranch and check for conflicts. From initRepo() we fetch all open gerrit-changes to the local branch-name "origin/${branchName}" to let this work as expected.
+
+- isBranchBehindBase()
+  The Implementation now checks all branches (not remote only) but includes the `origin/` prefix in the match. This way the fake-branches checked out from initRepo() return the correct answer, and it should work for the other platforms too (because `remotes/origin/$branchname`).
+
+- deleteBranch()
+  Fails always because remote branches are not exists in real. This should be no real problem.
```

Comment:
This sentence needs a rewrite.

Reply:
The last one only or the whole section?

The last sentence  could be changed to:

> Gerrit don't use branches, therefore `deleteBranch()` is unsupported and fails always silently.

But perhaps the complete part can/should be removed from the documentation and moved to a new Refactor-issue?

The Workarounds/Changes section is more or less a description of all my changes to the util/git module to get the Gerrit platform working. 
In my opinion it would be better to move these (all util/git) commands behind the platform interface. 
This way it could be possible later to integrate platforms without git too.

But that's a major refactoring and difficult to keep up-to-date because of the many places needed to change.

Reply:
We don't like to move away from our gitlayer again, we had that long ago, where every platform needs to implement and duplicate so much code. It wasn't maintainable, so we should avoid in future.

Reply:
moving the gitlayer behind an interface does not mean to duplicate the implementation for the existing platforms.

Some commands like `isBranchModified()` can't be answered thru git commands for the Gerrit platform.
I don't really like my workarounds there (registerBranch), but i see no other option then moving the function to the platform interface.

Reply:
We could abstract it behind the platform, with it defaulting to `git` if the platform doesn't provide the equivalent command

Reply:
> I tested this today on our internal Gerrit. First: really awesome stuff and very much looking forward to seeing it in action. Pretty much worked out-of-the-box in my tests.

Thanks for testing. Good to hear that it works.

> 
> Take the below please as small user feedback, not sure if that is in anyway feasible to easily implement or even needed :)
> 
> On open question we had, and might just be a documentation thing, is how to enable rebasing. Currently it still shows the checkbox, and as far as I know you can't tick/edit the comment on Gerrit? <img alt="image" width="804" src="https://user-images.githubusercontent.com/16856448/203557606-8435ebe6-b912-4554-8c52-6a506c7376d7.png">
> 
> It might be worth to add some documentation if that works and maybe even adjust the PR-content to mention how to do it to have the information self-contained.
> 

good point, after so much tests i've stopped reading the text inside these messages ;)

### at first to the message text
currently there is no option for a specific platform to influence the content. (correct me if i'm wrong)

Therefore documentation looks like the only option to translate the wording to the Gerrit context.
Some examples: 
 `**Schedule**: Branch creation...` -> `Change creation...`
`**Ignore**:  Close this PR...` -> `Vote with -2..`

### to the rebase part

currently i've enabled `platform.getRepoForceRebase() = true` for Gerrit. 
With the default renovate `rebaseWhen=auto` configuration this leads to automatically rebase of each Change if the baseBranch was updated. 
I like this behavior because each change will go thru CI (if attached to Gerrit).

But perhaps this leads to a behavior where it's impossible to disable these auto-rebase of changes and should be changed. 
I can get my wanted behavior with `rebaseWhen=behind-base-branch` even if getRepoForceRebase returns false. 
Comments? Suggestions?

But i don't see a real need for the "rebase checkbox" there. 
If it's not done automatically the developer can do this from Gerrit-UI....or is there a use-case?

If it's regardless needed, the only way of interacting with renovate is "reply" with a new message containing special commands. 
That's not trivial, because it's not possible to delete a message (after rebase was done).

Reply:
> We could abstract it behind the platform, with it defaulting to `git` if the platform doesn't provide the equivalent command

@rarkins Is this an requirement for this PR oder could be done after?

I've created #19065 for further discussion about the details.